ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh

```diff
@@ -1,117 +1,114 @@
-                grep -v -e ^default -e ^\$(id -u) /etc/passwd > "/tmp/passwd"
-                echo "\${USER_NAME:-default}:x:\$(id -u):0:\${USER_NAME:-default} user:\${HOME}:/sbin/nologin" >> "/tmp/passwd"
-                export LD_PRELOAD=libnss_wrapper.so
-                export NSS_WRAPPER_PASSWD=/tmp/passwd
-                export NSS_WRAPPER_GROUP=/etc/group
-            elif [[ -w /etc/passwd ]]; then
-                echo "\${USER_NAME:-default}:x:\$(id -u):0:\${USER_NAME:-default} user:\${HOME}:/sbin/nologin" >> "/etc/passwd"
-            else
-                echo "No nss wrapper, /etc/passwd is not writeable, and user matching this uid is not found."
-                exit 1
-            fi
-        fi
-      dest: "${SHARED_DIR}/fix-uid.sh"
-
-  - name: write Packet common configuration file
-    copy:
-      content: |
-        source "\${SHARED_DIR}/fix-uid.sh"
-
-        # Initial check
-        if [[ ! "\${CLUSTER_TYPE}" =~ ^packet.*$|^equinix.*$ ]]; then
-            echo >&2 "Unsupported cluster type '\${CLUSTER_TYPE}'"
-            exit 1
-        fi
-
-        IP=\$(cat "\${SHARED_DIR}/server-ip")
-        SSHOPTS=(-o 'ConnectTimeout=5' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -o 'ServerAliveInterval=90' -o LogLevel=ERROR -i "\${CLUSTER_PROFILE_DIR}/packet-ssh-key")
-
-        # Checkout packet server
-        for x in \$(seq 10) ; do
-            test "\${x}" -eq 10 && exit 1
-            ssh "\${SSHOPTS[@]}" "root@\${IP}" hostname && break
-            sleep 10
-        done
-      dest: "${SHARED_DIR}/packet-conf.sh"
+create_device
+wait_active
+wait_ssh
+
+cat <<EOF > "${SHARED_DIR}/fix-uid.sh"
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [ -x "\$(command -v nss_wrapper.pl)" ]; then
+        grep -v -e ^default -e ^\$(id -u) /etc/passwd > "/tmp/passwd"
+        echo "\${USER_NAME:-default}:x:\$(id -u):0:\${USER_NAME:-default} user:\${HOME}:/sbin/nologin" >> "/tmp/passwd"
+        export LD_PRELOAD=libnss_wrapper.so
+        export NSS_WRAPPER_PASSWD=/tmp/passwd
+        export NSS_WRAPPER_GROUP=/etc/group
+    elif [[ -w /etc/passwd ]]; then
+        echo "\${USER_NAME:-default}:x:\$(id -u):0:\${USER_NAME:-default} user:\${HOME}:/sbin/nologin" >> "/etc/passwd"
+    else
+        echo "No nss wrapper, /etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+EOF
+
+cat <<EOF > "${SHARED_DIR}/packet-conf.sh"
+source "\${SHARED_DIR}/fix-uid.sh"
+
+# Initial check
+if [ "\${CLUSTER_TYPE}" != "packet" ]; then
+    echo >&2 "Unsupported cluster type '\${CLUSTER_TYPE}'"
+    exit 1
+fi
+
+IP=\$(cat "\${SHARED_DIR}/server-ip")
+SSHOPTS=(-o 'ConnectTimeout=5' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -o 'ServerAliveInterval=90' -o LogLevel=ERROR -i "\${CLUSTER_PROFILE_DIR}/packet-ssh-key")
+
+# Checkout packet server
+for x in \$(seq 10) ; do
```

Comment:
On a second thought, probably this one is not required anymore (we already did an ssh availability test)?